src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -1,19 +1,71 @@
 import * as React from 'react';
+import * as API from '../../services/Api';
 import { RouteComponentProps } from 'react-router-dom';
+import PropTypes from 'prop-types';
 import NamespaceId from '../../types/NamespaceId';
 import CytoscapeLayout from '../../components/CytoscapeLayout/CytoscapeLayout';
+import { DropdownButton, MenuItem } from 'patternfly-react';
 
-export default function ServiceGraphPage(routeProps: RouteComponentProps<NamespaceId>) {
-  const namespace = routeProps.match.params.namespace;
+type ServiceGraphPageProps = {
+  availableNamespaces: { name: string }[];
+  graphNamespace: string;
+};
 
-  return (
-    <div className="container-fluid container-pf-nav-pf-vertical">
-      <div className="page-header">
-        <h2>Services Graph for namespace: {namespace} </h2>
-      </div>
-      <div style={{ height: 600 }}>
-        <CytoscapeLayout namespace={namespace} />
+export default class ServiceGraphPage extends React.Component<RouteComponentProps<NamespaceId>, ServiceGraphPageProps> {
```

Comment:
I was hoping we could keep the top level page stateless but we can work on that stuff later.

Reply:
Probably, moving out the dropdown to its own component will help to make it stateless again. Yes, let's work on that in the next sprint.

Reply:
Our state and props should always be typed. We seem to have lost type safety on the state. We should add that back next sprint.